src/ser-ext-ondemandDirective.ts

```diff
@@ -230,44 +231,44 @@ class OnDemandController implements ng.IController {
-                    let a = await this.app.sufficientRightsBookmark(["read", "delete", "update"]);
-                    if (!a) {
-                        this.state = ESERState.errorInsufficentRights;
-                        this.logger.warn("insufficient Rights for using bookmarks as connection type, please che you Security Rules");
-                        this.clearInterval();
-                        return;
-                    }
-                }
+        try {
+            const objectProperties = await model.getProperties()
+            const properties: IProperties = objectProperties.properties
 
-                if ((typeof (this.tempContentLibIndex) !== "undefined"
-                    && this.tempContentLibIndex !== res.properties.templateContentLibrary)
-                    || !this.checkIfTemplateExistsAsContent(res.properties.template)) {
-                    res.properties.template = null;
-                }
-                this.tempContentLibIndex = res.properties.templateContentLibrary;
-
-                if (isNull(res.properties.template)) {
-                    this.noPropertiesSet = true;
-                    this.state = ESERState.noProperties;
-                } else {
-                    this.noPropertiesSet = false;
-                    this.getStatus(this.taskId);
+            if (typeof(properties) !== "undefined" && typeof(properties.loglevel) !== "undefined") {
+                this.logger.setLogLvl(properties.loglevel)
+            }
+            await this.extractObjectProperties(properties)
+            if (properties.selection === ESelectionMode.bookmark) {
+                let a = await this.app.sufficientRightsBookmark(["read", "delete", "update"]);
+                if (!a) {
+                    this.state = ESERState.errorInsufficentRights;
+                    this.logger.warn("insufficient Rights for using bookmarks as connection type, please che you Security Rules");
+                    this.clearInterval();
+                    return;
                 }
-            })
-            .catch((error) => {
-                this.logger.error("ERROR in setter of model ", error);
-            });
+            }
+
+            if ((typeof (this.tempContentLibIndex) !== "undefined"
+            && this.tempContentLibIndex !== properties.templateContentLibrary)
+            || !this.checkIfTemplateExistsAsContent(properties.template)) {
+                properties.template = null;
+            }
+            this.tempContentLibIndex = properties.templateContentLibrary;
+
+            if (isNull(properties.template)) {
+                this.noPropertiesSet = true;
+                this.state = ESERState.noProperties;
+            } else {
+                this.noPropertiesSet = false;
+                this.getStatus(this.taskId);
+            }
+        } catch (error) {
+            this.logger.error("ERROR in fcn modelChanged() ", error);
+        }
```

Comment:
Dont know  it is a good idea to catch that error here and just log something out, i mean what if you want handle that error you simply cant anymore or is this not required anymore ?

Reply:
thanks alot for the comments, i will fix them. According to this issue, at this point it is ok, because it will fall back to default and i still see the error in the console. But will think about it to get a better workflow